great_expectations/core/batch.py

```diff
@@ -998,132 +1190,58 @@ def get_batch_request_from_acceptable_arguments(  # noqa: C901 - complexity 21
-                    'instead of "batch_filter_parameters". The "batch_identifiers" key on '
-                    'data_connector_query has been renamed to "batch_filter_parameters". Please update '
-                    'your code. Falling back on provided "batch_identifiers".'
-                )
-                batch_filter_parameters = batch_identifiers
-            elif batch_filter_parameters is None and batch_identifiers is None:
-                batch_filter_parameters = kwargs
-            else:
-                # Raise a warning if kwargs exist
-                pass
-
-            data_connector_query_params: dict = {
-                "batch_filter_parameters": batch_filter_parameters,
-                "limit": limit,
-                "index": index,
-                "custom_filter_function": custom_filter_function,
-            }
-            data_connector_query = IDDict(data_connector_query_params)
-        else:
-            # Raise a warning if batch_filter_parameters or kwargs exist
-            data_connector_query = IDDict(data_connector_query)
-
-        if batch_spec_passthrough is None:
-            batch_spec_passthrough = {}
-            if sampling_method is not None:
-                sampling_params: dict = {
-                    "sampling_method": sampling_method,
-                }
-                if sampling_kwargs is not None:
-                    sampling_params["sampling_kwargs"] = sampling_kwargs
-                batch_spec_passthrough.update(sampling_params)
-            if splitter_method is not None:
-                splitter_params: dict = {
-                    "splitter_method": splitter_method,
-                }
-                if splitter_kwargs is not None:
-                    splitter_params["splitter_kwargs"] = splitter_kwargs
-                batch_spec_passthrough.update(splitter_params)
+    # try to get a fluent batch request
+    if (datasource_name and data_asset_name) and not data_connector_name:
+        block_args = block_style_args()
+        if block_args:
+            raise ValueError(
+                f"Arguments: {', '.join(block_args)} are not supported for Fluent Batch Requests. Block-config Requests require a data connector name"
+            )
 
-        batch_request_as_dict: dict = {
-            "datasource_name": datasource_name,
-            "data_connector_name": data_connector_name,
-            "data_asset_name": data_asset_name,
-            "data_connector_query": data_connector_query,
-            "batch_spec_passthrough": batch_spec_passthrough,
-        }
+        result = _get_fluent_batch_request_class()(
+            datasource_name=datasource_name,
+            data_asset_name=data_asset_name,
+            options=batch_request_options,
+        )
+        return result
 
-    deep_filter_properties_iterable(
-        properties=batch_request_as_dict,
-        inplace=True,
+    # try to get a block batch request
+    return _get_block_batch_request(
```

Comment:
for block batch requests, we either error or return a value (where other methods may return `None`)